srcjs/shinyapp.js

```diff
@@ -375,6 +375,17 @@ var ShinyApp = function() {
     }
   };
 
+
+  // Narrows a scopeComponent -- an input or output object -- to one constrained
+  // by nsPrefix. Returns a new object with keys removed and renamed as
+  // necessary.
+  function narrowScope(scopeComponent, nsPrefix) {
+    return Object.keys(scopeComponent)
+      .filter(k => k.startsWith(nsPrefix))
+      .map(k => ({[k.substring(nsPrefix.length)]: scopeComponent[k]}))
+      .reduce((obj, pair) => Object.assign(obj, pair), {});
```

Comment:
`Object.assign` isn't supported by IE at all, and it's not polyfilled, so we should use `$.extend()` instead.